modules/mono/mono_gd/gd_mono.cpp

```diff
@@ -365,6 +375,40 @@ godot_plugins_initialize_fn try_load_native_aot_library(void *&r_aot_dll_handle)
 
 } // namespace
 
+bool GDMono::should_initialize() {
+#ifdef TOOLS_ENABLED
+	String project_assembly_name = (String)ProjectSettings::get_singleton()->get_setting("dotnet/project/assembly_name");
+	if (project_assembly_name.is_empty()) {
+		String appname = GLOBAL_GET("application/config/name");
+		String appname_safe = OS::get_singleton()->get_safe_dir_name(appname);
+		if (appname_safe.is_empty()) {
+			appname_safe = "UnnamedProject";
+		}
+		project_assembly_name = appname_safe;
+	}
```

Comment:
Should be able to use `path::get_csharp_project_name` here.

```suggestion
	String project_assembly_name = path::get_csharp_project_name();
```